File: cctv/server-java/src/main/java/com/example/clinicserver/model/Attendance.java
package com.example.clinicserver.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Map;

@Document(collection = "attendance")
public class Attendance {

    @Id
    private String id;

    private String date; // ISO date string yyyy-MM-dd

    private String section;

    // key: roll number, value: status (e.g., "present" / "absent" / "sick")
    private Map<String, String> records;

    public Attendance() {}

    public Attendance(String date, String section, Map<String, String> records) {
        this.date = date;
        this.section = section;
        this.records = records;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getSection() {
        return section;
    }

    public void setSection(String section) {
        this.section = section;
    }

    public Map<String, String> getRecords() {
        return records;
    }

    public void setRecords(Map<String, String> records) {
        this.records = records;
    }
}

File: cctv/models/Attendance.js
import mongoose from 'mongoose';

const AttendanceSchema = new mongoose.Schema(
  {
    date: { type: String, index: true }, // yyyy-mm-dd
    section: { type: String, index: true },
    records: { type: Object }, // map of roll -> status ('present'|'absent'|'sick')
    by: { type: String }
  },
  { timestamps: true }
);

const Attendance = mongoose.models.Attendance || mongoose.model('Attendance', AttendanceSchema);
export default Attendance;
